### PR TITLE
switch to github actions

### PR DESCRIPTION
**Pull request by marcoow**
…as it will hopefully speed things up significantly as simplabs/ember-test-selectors#443 did.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - 'v*'
+      - 2.x
+      - /^\d+\.\d+\.\d+/
```
Comment (Turbo87) [inline .github/workflows/ci.yml]: maybe we should standardize on one way to name our version branches 😅

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      - name: lint
+        run: yarn lint
+
+  tests:
+    name: Tests
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        test-suite: ["test:one ember-3.0", "test:one ember-lts-3.4", "test:one ember-lts-3.8", "test:one ember-lts-3.12", "test:one ember-release", "test:one ember-beta", "test:node", "test:fastboot", "test:one ember-canary"]
```
Comment (Turbo87) [inline .github/workflows/ci.yml]: is there a way to format this differently to make it more readable?

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: install dependencies
+        run: yarn install
+
+      - name: Deploy
+        env:
+          NOW_AUTH_TOKEN: ${{ secrets.NOW_TOKEN }}
+        run: |
+          now deploy --token=$NOW_AUTH_TOKEN --scope=e-simple-auth && now alias --token=$NOW_AUTH_TOKEN --scope=e-simple-auth
+          cd server && now deploy --token=$NOW_AUTH_TOKEN --scope=e-simple-auth && now alias --token=$NOW_AUTH_TOKEN --scope=e-simple-auth
```
Comment (Turbo87) [inline .github/workflows/deploy.yml]: do we need both?

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12.x
+
+      - run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (Turbo87) [inline .github/workflows/release.yml]: I assume you've set it in the github repo secrets section?

File: .travis.yml [hunk 5]
```diff
@@ -1,84 +0,0 @@
...
-    - stage: 'Basic Tests'
-      name: 'Lint'
-      env: SCRIPT=lint
-
-    - name: 'Fixed Dependencies'
-      env: SCRIPT="test"
-
-    - name: 'Floating Dependencies'
-      env: SCRIPT="test"
-      install: yarn install --no-lockfile
```
Comment (Turbo87) [inline .travis.yml]: we don't seem to test this case anymore?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (marcoow) [reply .github/workflows/ci.yml]: `/^\d+\.\d+\.\d+/` is for the release tags though - for ember-test-selectors we use `'v*'` but we're not using the prefix here yet. I copied it anyway as we might want to streamline the tags across our repositories. `2.x` is mainly relevant now as `master` contains all the changes for `3.0` already.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (marcoow) [reply .github/workflows/ci.yml]: yes, done

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (marcoow) [reply .github/workflows/deploy.yml]: Yes, one deploys the Ember app, the other one deploys the mock server.

File: .github/workflows/release.yml (hunk 4 above)
Comment (marcoow) [reply .github/workflows/release.yml]: yes 👍

File: .travis.yml (hunk 5 above)
Comment (marcoow) [reply .travis.yml]: added back in

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Turbo87) [reply .github/workflows/ci.yml]: then it should be under `tags`, not `branches` 🤔

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (Turbo87) [reply .github/workflows/deploy.yml]: it seems to be the exact same call. will they be deployed to different locations? 🤔

Comment (marcoow) [reply .github/workflows/deploy.yml]: Yes, one has `"name": "simple-auth-server",` in the `package.json`, the other is `"name": "ember-simple-auth",` - see the `cd server`

Comment (Turbo87) [reply .github/workflows/deploy.yml]: I see 👍

File: .github/workflows/ci.yml (hunk 1 above)
Comment (marcoow) [reply .github/workflows/ci.yml]: You're right - we don't need this at all actually. I also renamed the `2.x` branch to `v2`.

**Comment by marcoow** [comment]
🎉  Once we fix the FastBoot tests so we don't run into a 10 minute timeout anymore, this will make the build dramatically faster!

**Pull request merged**